src/build/specialize_types/IR.zig

```diff
@@ -48,72 +55,87 @@ pub fn deinit(self: *Self) void {
     self.when_branches.deinit(self.env.gpa);
 }
 
-/// todo
+/// Represents a Roc type e.g. `U64`, `Str`, `List(U64)`
 pub const Type = union(enum) {
+    /// e.g. Int, Bool, Str
     primitive: types_module.Primitive,
+    /// Holds unknown Roc type for use with platform or can be used to improve perf in rare cases roc-lang.org/builtins/Box
     box: Type.Idx,
+    /// e.g. List(U64)
     list: Type.Idx,
+    /// ??? Record?
     @"struct": Type.NonEmptySlice,
+    /// e.g. `[Red, Yellow, Green]`
     tag_union: Type.NonEmptySlice,
     func: struct {
+        /// ???
         ret_then_args: Type.NonEmptySlice,
     },
-    /// todo
+    /// list of Type
     pub const List = collections.SafeList(@This());
-    /// todo
+    /// Index into Type.List
     pub const Idx = List.Idx;
-    /// todo
+    /// Sublist reference of Type.List
     pub const Slice = List.Slice;
-    /// todo
+    /// Sublist reference into Type.List that is guaranteed to be non-empty
     pub const NonEmptySlice = List.NonEmptySlice;
 };
 
-/// todo
+/// Represents a Roc expression
 pub const Expr = union(enum) {
+    /// e.g. `x = 1`
     let: Def,
+    /// e.g. "abc"
     str: StringLiteral,
+    /// e.g. 123
     number: base.Literal.Num,
+    /// e.g. [1, 2, 3]
     list: struct {
         elem_type: Type.Idx,
         elems: Expr.Slice,
     },
+    /// e.g. `pf.Stdout` or `x`
     lookup: struct {
         ident: Ident.Idx,
         type: Type.Idx,
     },
-
+    /// e.g `fibo(5)`
     call: struct {
         fn_type: Type.Idx,
+        // ???
         fn_expr: Expr.Idx,
         args: Expr.Typed.Slice,
     },
-
+    /// e.g. `|x| x * 2`
     lambda: struct {
         fn_type: Type.Idx,
         arguments: Pattern.Typed.Slice,
         body: Expr.Idx,
         recursive: base.Recursive,
     },
-
+    /// ??? What's this?
     unit,
-
+    /// ??? What's this?
     @"struct": Expr.NonEmptySlice,
-
+    /// Should this be called record access?
     struct_access: struct {
         record_expr: Expr.Idx,
         record_type: Type.Idx,
         field_type: Type.Idx,
         field_id: Ident.Idx,
     },
-
+    /// e.g. `Ok(1)`
     tag: struct {
+        /// ??? Numeric representation of the tag variant? How to find the name of the tag?
         discriminant: u16,
```

Comment:
Tag unions get compiled to [discriminated unions](https://en.wikipedia.org/wiki/Tagged_union), which means in memory they have two fields:
- A single number (probably a byte) that represents which value is being held, this is called the discriminant
- A union with memory size that holds as much memory is required to hold the payload of the largest tag

The idea with these would be that we'd sort them alphabetically and store them in that order, but that might change to something simpler to start with, to be seen.